frontend/ui/widget/bookmapwidget.lua

```diff
@@ -1341,13 +1346,16 @@ function BookMapWidget:showMenu()
     }
     UIManager:show(button_dialog)
 end
+
+function BookMapWidget:onShowMenu()
+    self:showMenu()
+end
```

Comment:
trying to rewrite this to something like,
```lua
function BookMapWidget:onShowMenu()
    if not self.ui.button_dialog then
        self:showMenu()
    else
        [HERE]
    end
end
```
but nothing [HERE] works (at least nothing I've tried), wondering what is it that I am doing wrong... (my life's theme ;) )

Reply:
You'd need to keep a reference to the thing passed to `UIManager:show(button_dialog)` (which is currenly a local variable, and not at all the one you think you would get with `self.ui.button_dialog`.
So, saving it as `self.menu_button_dialog`.
Then, in the else branch (but then, call it onToggleMenu), you would do:
```lua
UIManager:close(self.menu_button_dialog)
self.menu_button_dialog = nil -- you need to make sure this keeps the real state, and that anything that closes it reset this to nil
```

Reply:
> and not at all the one you think you would get with self.ui.button_dialog.

so what am I getting here? 

something like this? still won't work

```lua
function BookMapWidget:onToggleMenu()
    local menu_button_dialog = UIManager:show(button_dialog)
    if not self.menu_button_dialog then
        self:showMenu()
    else
        UIManager:close(self.menu_button_dialog)
        self.menu_button_dialog = nil
    end
end
```

the back button can also close the dialog but I am not sure how to go around making sure that it resets `self.menu_button_dialog = nil` when pressed, as it is mapped to Close the whole widget, and I think it inherits the behaviour of closing focus first from elsewhere.

I think I might just leave it as it currently is.

Reply:
At the bottom of BookMapWidget:showMenu(), add `return button_dialog`.
Then:
```lua
function BookMapWidget:onShowMenu()
    if not self.menu_button_dialog then
        self.menu_button_dialog = self:showMenu()
    else
        UIManager:close(self.menu_button_dialog)
        self.menu_button_dialog = nil
    end
end
```

Reply:
that does not work either.

Reply:
Oh, right. Indeed, when the ButtonDialog is displayed, it is no longer this BookMarpWidget that get the events, it's that ButtonDialog.
So, closing with the same key (F1 on the emulator) should be handled in ButtonDialog, where there are already keys for closing.
Dunno if that new key should always be enabled, or depend on an option, or on other existing options defining some behaviour where it would make sense to allow closing with that new key.

Reply:
I guess this question is for @Frenzie :)

Reply:
Even hasFewKeys has a menu button ftr, so possibly all three on hasFewKeys (the left only added for hasFewKeys consistenty).